src/viperleed/calc/files/input_reader.py

```diff
@@ -0,0 +1,153 @@
+"""Module input_reader of viperleed.calc.files.
+
+This module is a more abstract version of a settings file reader. It is based on
+@michele-riva's work on the PARAMETERS file reader, which in turn is based on
+the old readPARAMETERS from @fkraushofer.
+
+Defines context-manager, iterator classes for reading parameters from a file.
+"""
+
+__authors__ = (
+    'Alexander M. Imre (@amimre)',
+    'Michele Riva (@michele-riva)',
+    )
+__copyright__ = 'Copyright (c) 2019-2024 ViPErLEED developers'
+__created__ = '2024-10-03'
+__license__ = 'GPLv3+'
+
+from abc import abstractmethod
+from io import TextIOBase
+from collections.abc import Iterator
+from contextlib import AbstractContextManager
+import logging
+from pathlib import Path
+
+_LOGGER = logging.getLogger(__name__)
+
+# TODO: Add subclasses for the InputReaders as suggested by @michele-riva in
+#       PR #266. E.g. DelimitedFileReader, BlockFileReader, etc.
+
+class ShouldSkipLineError(Exception):
+    """Exception raised when a line's contents should be skipped."""
+
+
+class InputReader(Iterator):
+    """Common base class for all input readers."""
+
+    def __init__(self, noisy=True) -> None:
+        self.noisy = noisy
+        super().__init__()
+
+    def __next__(self):
+        """Return the next understandable information in the file."""
+        for line in self.stream:
+            try:
+                return self._read_one_line(line)
+            except ShouldSkipLineError:
+                # Intentional sub-debug logging level as this is expected
+                # behavior and may spam the logs.
+                _LOGGER.log(level=5,
+                            msg=f'Skipping line in input file: "{line}".')
+                continue
+        raise StopIteration
+
+    @property
+    @abstractmethod
+    def stream(self):
+        """Return the input stream."""
+        pass
+
+    @abstractmethod
+    def _read_one_line(self, line):
+        """Return understandable information from `line`.
+
+        This method is guaranteed to be called once on each
+        line read from the file.
+
+        Parameters
+        -------------
+        line : str
+            A single line read from `self._file_obj.`
+
+        Returns
+        --------
+        info : object
+            Understandable information read from `line`. This is the
+            same object returned whenever this reader is iterated over.
+
+        Raises
+        -------
+        ShouldSkipLineError
+            If `line` does not contain any valuable information that is
+            worth returning while iterating over the file.
+        """
+        pass
+
+
+class InputStreamReader(InputReader):
```

Comment:
It may not be apparent from reading the code that this is still an ABC. You can signal that by

```python
from abc import ABC

class InputStreamReader(ABC, InputReader):
```